src/app/clusters/scenes-server/scenes-server.cpp

```diff
@@ -0,0 +1,993 @@
+    }
+}
+void ScenesServer::RecallScene(FabricIndex aFabricIx, EndpointId aEndpointId, GroupId aGroupId, SceneId aSceneId)
+{
+    VerifyOrReturn(EMBER_ZCL_STATUS_SUCCESS == Attributes::SceneValid::Set(aEndpointId, false));
+
+    Optional<DataModel::Nullable<uint16_t>> transitionTime;
+
+    if (CHIP_NO_ERROR == RecallSceneParse(aFabricIx, aEndpointId, aGroupId, aSceneId, transitionTime, mGroupProvider))
+    {
+        Attributes::SceneValid::Set(aEndpointId, true);
+    }
+}
+
+void ScenesServer::HandleAddScene(HandlerContext & ctx, const Commands::AddScene::DecodableType & req)
+{
+    AddSceneParse<Commands::AddScene::DecodableType, Commands::AddSceneResponse::Type>(ctx, req, mGroupProvider);
+}
+
+void ScenesServer::HandleViewScene(HandlerContext & ctx, const Commands::ViewScene::DecodableType & req)
+{
+    ViewSceneParse<Commands::ViewScene::DecodableType, Commands::ViewSceneResponse::Type>(ctx, req, mGroupProvider);
+}
+
+void ScenesServer::HandleRemoveScene(HandlerContext & ctx, const Commands::RemoveScene::DecodableType & req)
+{
+    Commands::RemoveSceneResponse::Type response;
+
+    uint16_t endpoinTableSize = 0;
+    ReturnOnFailure(AddResponseOnError<Commands::RemoveSceneResponse::Type>(
+        ctx, response, Attributes::SceneTableSize::Get(ctx.mRequestPath.mEndpointId, &endpoinTableSize)));
+
+    // Get Scene Table Instance
+    scenes::SceneTable<scenes::ExtensionFieldSetsImpl> * sceneTable =
+        chip::scenes::GetSceneTableImpl(ctx.mRequestPath.mEndpointId, endpoinTableSize);
+
+    // Response data
+    response.groupID = req.groupID;
+    response.sceneID = req.sceneID;
+
+    // Scene Table interface data
+    SceneTableEntry scene(SceneStorageId(req.sceneID, req.groupID));
+
+    // Verify Endpoint in group
+    VerifyOrReturn(nullptr != mGroupProvider);
+    if (0 != req.groupID &&
+        !mGroupProvider->HasEndpoint(ctx.mCommandHandler.GetAccessingFabricIndex(), req.groupID, ctx.mRequestPath.mEndpointId))
+    {
+        response.status = to_underlying(Protocols::InteractionModel::Status::InvalidCommand);
+        ctx.mCommandHandler.AddResponse(mPath, response);
+        return;
+    }
+
+    //  Gets the scene from the table
+    ReturnOnFailure(AddResponseOnError<Commands::RemoveSceneResponse::Type>(
+        ctx, response, sceneTable->GetSceneTableEntry(ctx.mCommandHandler.GetAccessingFabricIndex(), scene.mStorageId, scene)));
+
+    // Remove the scene from the scene table
+    ReturnOnFailure(AddResponseOnError<Commands::RemoveSceneResponse::Type>(
+        ctx, response, sceneTable->RemoveSceneTableEntry(ctx.mCommandHandler.GetAccessingFabricIndex(), scene.mStorageId)));
+
+    // Update Attributes
+    uint8_t scene_count = 0;
+    ReturnOnFailure(AddResponseOnError<Commands::RemoveSceneResponse::Type>(
+        ctx, response, sceneTable->GetFabricSceneCount(ctx.mCommandHandler.GetAccessingFabricIndex(), scene_count)));
+
+    Attributes::SceneCount::Set(ctx.mRequestPath.mEndpointId, scene_count);
```

Comment:
Again (and other places above): if we are reading this and RemainingCapacity via AttributeAccessInterface, then why do we bother with storing them?